src/main/java/com/trolley/trolley/Invoice.java

```diff
@@ -0,0 +1,311 @@
+        return updatedAt;
+    }
+
+    public void setUpdatedAt(String updatedAt) {
+        this.updatedAt = updatedAt;
+    }
+
+    public Amount getTotalAmount() {
+        return totalAmount;
+    }
+
+    public void setTotalAmount(Amount totalAmount) {
+        this.totalAmount = totalAmount;
+    }
+
+    public Amount getPaidAmount() {
+        return paidAmount;
+    }
+
+    public void setPaidAmount(Amount paidAmount) {
+        this.paidAmount = paidAmount;
+    }
+
+    public Amount getDueAmount() {
+        return dueAmount;
+    }
+
+    public void setDueAmount(Amount dueAmount) {
+        this.dueAmount = dueAmount;
+    }
+
+    public List<String> getTags() {
+        return tags;
+    }
+
+    public void setTags(List<String> tags) {
+        this.tags = tags;
+    }
+
+    public List<InvoiceLine> getLines() {
+        return lines;
+    }
+
+    public void setLines(List<InvoiceLine> lines) {
+        this.lines = lines;
+    }
+
+    public String getRecipientId() {
+        return recipientId;
+    }
+
+    public void setRecipientId(String recipientId) {
+        this.recipientId = recipientId;
+    }
+
+    /**
+     * Factory method to generate Invoice object from a String representation of a valid Invoice JSON 
+     * @param invoiceStr
+     * @return
+     * @throws IOException
+     */
+    public static Invoice invoiceFactory(final String invoiceStr) throws IOException {
+        final ObjectMapper mapper = new ObjectMapper();
+        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        final JsonNode node = mapper.readTree(invoiceStr);
+        String invoiceId = node.get("invoice").get("id").asText();
+        try{
+            final Invoice invoice = (Invoice)mapper.readValue(node.get("invoice").traverse(), (Class)Invoice.class);
+            return invoice;
+        }catch(Exception e){
+            e.printStackTrace();
+        }
+        return new Invoice(invoiceId, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null);
+    }
+
+    /**
+     * Factory method to create a List of Invoices
+     * @param data
+     * @return
+     * @throws IOException
+     */
+    public static Invoices invoiceListFactory(final String invoicesStr) throws IOException {
+        final ObjectMapper mapper = new ObjectMapper();
+        final JsonNode node = mapper.readTree(invoicesStr);
+        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        final Object invoice = mapper.readValue(node.get("invoices").traverse(), (Class)Object.class);
+        final List<Invoice> parsedInvoices = (List<Invoice>)invoice;
+        final List<Invoice> invoiceObjs = new ArrayList<Invoice>();
+
+        final Meta meta = (Meta)mapper.readValue(node.get("meta").traverse(), (Class)Meta.class);
+        
+        //Creating List<Invoice> object to return
+        for (int i = 0; i < parsedInvoices.size(); ++i) {
```

Comment:
I think you can achieve same results with
```suggestion
        for (Object invoice : parsedInvoices) {
        final Invoice pojo = (Invoice)mapper.convertValue((Object)invoice, (Class)Invoice.class);
            invoiceObjs.add(pojo);
        }
```

Reply:
Good catch again. Legacy code.
I'll update this implementation, slating the older ones for the next update.